src/include/miopen/generic_search.hpp

```diff
@@ -298,6 +300,50 @@ auto GenericSearch(const Solver s,
 }
 
 std::size_t GetTuningIterationsMax();
+std::chrono::milliseconds GetTuningTimeMax(); // returns the max allowed time in milliseconds
+std::size_t GetTuningThreadsMax();
+
+template <typename PerformanceConfig, typename Solver, typename Context>
+void CompileAgent(size_t thread_index,
+                  size_t total_threads,
+                  const Solver& s,
+                  const Context& context_,
+                  std::vector<PerformanceConfig>& data,
+                  ThreadSafeQueue<std::tuple<PerformanceConfig, ConvSolution, bool>>& comp_queue)
+{
+    const auto start_time =
+        std::chrono::time_point_cast<std::chrono::milliseconds>(std::chrono::system_clock::now());
+    const auto data_size          = data.size();
+    const auto time_budget        = GetTuningTimeMax();
+    auto context                  = context_; // Not sure if context is thread safe
+    context.is_for_generic_search = true;
+    const auto& profile_h         = context.GetStream();
+    // start the counter
+    for(auto idx = thread_index; idx < data_size; idx += total_threads)
+    {
+        // Check if we are out of time
+        const auto current_time = std::chrono::time_point_cast<std::chrono::milliseconds>(
+            std::chrono::system_clock::now());
```

Comment:
`std::chrono::system_clock` may not be monotonic, it is better to use `std::chrono::steady_clock`

Reply:
Thanks for the tip, done